### ISSUE TITLE
What High-end hardware for faster training?

### ISSUE THREAD

**Issue by MiloMindbender** [tdrussell/diffusion-pipe#213]
So if I need to train LORAs faster.  What type of high-end server hardware will work?  Bigger GPUs?  More of them?

If I want to take a training process that runs for hours on a 4090, what would it take to get the time down to minutes?  Would having access to an 8 gpu box or large scale AI server help or is there a point where just throwing more hardware at the job won't speed things up?

The goal here is to reduce training time as much as possible to allow researchers to run more experiments with different settings, captioning styles...etc to see what generates the best results.

What I want to do is reduce training time so someone can make a LORA, change a few things, remake it, see if it's better...etc.  ie: try a bunch of experiments without having to wait hours between results.

**Comment by alisson-anjos** [comment]
In my opinion, you can increase the Learning Rate so that you reach the optimum point faster, and with that you will need fewer steps to have some result, the issue is that this can increase the chance of overfitting, another thing would be to train in specific blocks, instead of training all the blocks of the model, for example, in Flux some people train only blocks 7, 14 and 20 in some cases, the training tends to be faster and more accurate in the blocks that really make sense for the type of lora you are training, another way is to try to use GPUs with more CUDA cores to process as many samples per second as possible.

**Comment by Basedbase-ai** [comment]
Changing the LR is not a one-size-fits-all solution you can very easily overshoot your LR making it too high and end up with a glitchy lora that does not perform well. You cant train extremely fast and spreading the training over many GPU's slows things down if you don't need the extra VRAM. If your offloading to system ram due to insufficient VRAM multi GPU's help with speed some by not having to offload to slower system memory. Some of the fastest things to train with would be something like an H100 but it will still take many many hours to get a good result. Complex subjects are more difficult to train (animated things in the case of wan 2.1 are complex for it to learn) and requires experimenting with LR, batch size etc. to get the best result. High LR's are also bad for the text encoder and cause it to overfit well before the Unet is trained fully and cause bad prompt adherence or even just straight black video outputs. All about balance and unfortunately at least for models like Wan 2.1 its quite complex with not so clear answers.

**Comment by tdrussell** [comment]
>Would having access to an 8 gpu box or large scale AI server help

Yes. With multiple GPUs in a data parallel configuration, you are increasing the global batch size with the same total time per step. With relatively small data parallel rank (e.g. 8) and a properly tuned learning rate, the speed at which the model learns in terms of wall clock time should scale approximately linearly with number of GPUs. 8xH100 will train to convergence about 8 times faster than 1xH100.

Keep in mind that using diffusion-pipe with multiple machines might be tricky. Renting 8xH100 is about as fast as you can possibly get without too much difficulty.

If you want to build something yourself for cheaper, and the VRAM per GPU isn't a limitation, you can go up to about 8x4090. There is some info out there of people building machines like this. Personally I use a 4x4090 machine.
